Add support for the style tag

tneotia opened the pull request:
Fixes #593 and sort of addresses #207 (we still don't support all the styles given in the example in that issue).

![image](https://user-images.githubusercontent.com/50850142/116828280-aef3dd80-ab6b-11eb-84a8-5671120783d0.png)


<details><summary> Test HTML </summary>

```html
<html>
<head>
<style>
body {
  background-color: lightblue;
}
</style>
<style>
h1 {
  color: white;
  text-align: center;
}
</style>
<style>
p {
  font-size: 20px;
}
</style>
</head>
<body>

<h1>My First CSS Example</h1>
<p style="color: blue;">This is a paragraph.</p>

</body>
</html>
```

</details>

~~Should we add a parameter to the `Html` widget to allow users to pass the external CSS as a string separate from the HTML data? Don't know if that would be useful but it wouldn't be difficult to implement at all.~~ Done via `Style.fromCSS` function.

 ~~Also for some reason adding `<head>` to the document adds a space at the top of the doc, I don't know if the changes made in #646 will help that but I don't think so.~~ Fixed

erickok commented:
As for the <head> tag occupying visible space, maybe render it explicitly as empty Container()?

tneotia commented:
> As for the tag occupying visible space, maybe render it explicitly as empty Container()?

It should be rendered as `EmptyContentElement` (as far as I can tell) which gets removed from the tree because it isn't listed as an element we render, which is why I'm confused it takes visible space.

tneotia commented:
Fixed the empty space issue with the recent commit - for some reason the `<html>` tag was rendered as a `TextContentElement` with `" "` as the text, so it didn't get flagged and removed as an "empty" `TextContentElement`.

erickok commented on lib/html_parser.dart:
```diff
@@ -66,15 +72,20 @@ class HtmlParser extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    Map<String, Map<String, List<css.Expression>>> declarations = _getExternalCSSDeclarations(htmlData.getElementsByTagName("style"), onCSSParseError);
```
Here too, camelCase please, so `_getExternalCssDeclarations`.

erickok reviewed (approved):
Approved  but could you change all naming to camelCase?

tneotia commented:
@erickok done.
